web/static/web/css/header.css

```diff
@@ -1,144 +1,332 @@
-.nav {
+#header {
+    --logo-padding: 30px;
+    --logo-width: 360px;
+    --logo-min-width: 120px;
+    --logo-space: minmax(var(--logo-min-width), var(--logo-width));
+
+    --burger-margin: 2em;
+
+    --header-height: 120px;
+
+    /* nav */
+    --nav-space-between: 2em;
+    --bubble-background-extra-width: calc(1.5 * var(--nav-space-between));
+    --dur: 0.15s;
+    --font-size: 1.25em;
+
     display: grid;
-    grid-template-columns: minmax(120px, 1fr) 3fr;
-    grid-template-rows: 120px;
+    grid-template-columns: 1fr auto 1fr;
     width: 100%;
-    height: 120px;
-    background-color: rgb(33, 43, 52);
+    height: var(--header-height);
+    background-color: rgb(34, 43, 52);
 }
 
-.nav .menu {
-    align-self: center;
-    margin: 0;
-    padding: 0 30px;
+#header .nav-item .nav-item-text,
+#nav-user-dropdown .dropdown-item .dropdown-item-text,
+.lang-link .item .text {
+    color: #DCDDDE;
 }
 
-.nav .logo {
+#header .nav-item:hover .nav-item-text,
+#nav-user-dropdown .dropdown-item:hover .dropdown-item-text,
+.lang-link .item:hover .text {
+    color: #FFF;
+}
+
+#header div.logo-centering {
+    display: none;
+}
+
+#header .logo {
+    display: grid;
+    grid-template-columns: var(--logo-padding) var(--logo-space) minmax(var(--nav-space-between), var(--logo-padding));
     align-self: center;
-    padding: 0 30px;
 }
 
-.nav .logo img {
+#header .logo img {
+    margin: auto;
+    max-height: 50px;
     object-fit: contain;
 }
 
-.limg {
+#header .logo .limg.large {
+    display: block;
+}
+
+#header .logo .limg.medium {
     display: none;
 }
 
-.limg.large {
-    display: inline;
+#header .logo .limg.small {
+    display: none;
 }
 
-.burger {
+#header #burger {
```

Comment:
That's true. I did it purely because I thought it increased readability, like the (unnecessarily) long chained selectors elsewhere in the stylesheet, because you can get an overview of which elements are inside which just by reading the selectors, and because I think it's easier to find where the properties for a specific element is located as I scroll through the stylesheet.

For example, if I was trying to find `.item` inside `#nav` inside `#header`, it would be quicker to be able to scroll down until I found `#header`, then `#header #nav`, and finally a little further to find `#header #nav .item`. I guess the alternative would be to sort the selectors alphabetically, or to only include the necessary selectors and place them in the same order as they appear in the HTML, and relying on either remembering that exact order to find a specific selector, or using Ctrl+F - which I guess is an absolutely feasible way of doing it :p

I do, however, get kind of blind to the readability after having read the same code over and over, so I'm unsure whether it's as readable as I initially thought, now that you pointed it out.

Reply:
I get what you are trying to achieve. However, I see a couple of problems.

- Duplication of code. The same X number of elements follow each other all the time. Meaning that you have these long chains of selectors all over the place (much code). The code is also kind of duplicated across HTML and CSS.
- A simple HTML change may require changes to most of the CSS, if it changes some of the ordering of the elements. Rather than only requiring changes to a couple of selectors.
- Long selectors makes the code look more complex. There is a reason why one usually set a limit to the length of line of code.

Personally, I find it just as easy to find the correct location in the CSS file if you instead group selectors based on their elements. E.g. everything related to `#nav` is grouped. Then everything related to `.item` is grouped and so on. That is, group selectors based on which class/id they change. 

I think sorting selectors alphabetically would be much harder to use. Especially due to the nature of CSS selectors. In which selectors for a single element are not necessarily grouped together alphabetically.

Reply:
Grouping selectors is basically what I meant when I said "place them in the same order as they appear in the HTML" :p I agree with that last point on sorting alphabetically. Other than that, you bring up some good points! Will change accordingly.